examples/ssh/nginx_stats.js

```diff
@@ -0,0 +1,316 @@
+
+/**
+ * this driver is using ssh for the target server
+ * the ssh user should have the sudo privilege to access log files
+ * this driver is tested under nginx version: nginx/1.14.0 (Ubuntu)
+ */
+
+var _var = D.device.createVariable;
+var validation_cmd = "ls -al";
+var cpu_info_cmd = "lscpu | grep -ie '^cpu(s):' -ie '^thread(s) per core:' | awk '{print $NF}'";
+var nginx_status_stats_cmds = [
+    "STATUS=$(systemctl status nginx)",
+    "echo status:Status:$(echo \"$STATUS\" | grep -i active | awk '{print $2 $3}')",
+    "PID=$(echo \"$STATUS\" | grep PID | awk '{print $3}')",
+    "echo pid:PID:$PID",
+    "echo memory:Memory usage:$(echo $PID | sudo xargs pmap | tail -1 | awk '{print $2}')",
+    "echo tasks:Tasks:$(echo \"$STATUS\" | grep -i tasks | awk '{print $2}')",
+    "echo tasks_limit:Tasks limit:$(echo \"$STATUS\" | grep -i tasks | awk '{print substr($4,1,length($4)-1)}')",
+    "echo up_from:Up from:$(echo \"$STATUS\" | grep -i active | awk '{print \"\\x27\" $6 \" \" $7 \" \" substr($8,1,length($8)-1) \"\\x27\" \" +\\\"%s\\\"\"}' | xargs date -d)date",
+];
+var nginx_system_config_cmd = "sudo systemctl show nginx --no-pager";
+var nginx_root_conf_file_location = "/etc/nginx/nginx.conf";
+
+function build_queries_for_log_file(id, logFile) {
+    var command = "last_hour=$(date +\"%d/%b/%Y:%H\" -d \"$DATE -1 hour\");CONTENT=$(sudo grep $last_hour " + logFile + " | cut -d \" \" -f 9)";
+    return [
+        command,
+        "echo " + id + ":all_requests:All Requests:$(echo \"$CONTENT\" | wc -l)",
+        "echo " + id + ":successfull_requests:Successfull Requests:$(echo \"$CONTENT\" | grep -E 2.. | wc -l)",
+        "echo " + id + ":redirected_requests:Redirected Requests:$(echo \"$CONTENT\" | grep -E 3.. | wc -l)",
+        "echo " + id + ":error_requests:Error Requests:$(echo \"$CONTENT\" | grep -E \"4..|5..\" | wc -l)"
+    ];
+}
+
+var ssh_config = {
+    username: D.device.username(),
+    password: D.device.password(),
+    port: 27123,
```

Comment:
I assume this port is rarely common for SSH
Best to define the default in the driver (22)